pkgs/development/cuda-modules/buildRedist/default.nix

```diff
@@ -0,0 +1,484 @@
+                attrNames finalAttrs.passthru.outputNameVarFallbacks
+              ) == [ ];
+          }
+          {
+            message = "outputToPatterns is a super set of expectedOutputs";
+            assertion =
+              subtractLists finalAttrs.passthru.expectedOutputs (attrNames finalAttrs.passthru.outputToPatterns)
+              == [ ];
+          }
+          {
+            message = "propagatedBuildOutputs is a subset of outputs";
+            assertion = subtractLists finalAttrs.outputs finalAttrs.propagatedBuildOutputs == [ ];
+          }
+        ]
+        ++ brokenAssertions;
+
+        # platformAssertions :: [Attrs]
+        # Used by mkMetaBadPlatforms to set `meta.badPlatforms`.
+        # Example: Broken on a specific system when some condition is met, like targeting Jetson or
+        # a required package missing.
+        # NOTE: Use this when a failed assertion means evaluation can fail!
+        platformAssertions =
+          let
+            isSupportedRedistSystem = _redistSystemIsSupported hostRedistSystem finalAttrs.passthru.supportedRedistSystems;
+          in
+          [
+            {
+              message = "src is null if and only if hostRedistSystem is unsupported";
+              assertion = (finalAttrs.src == null) == !isSupportedRedistSystem;
+            }
+            {
+              message = "hostRedistSystem (${hostRedistSystem}) is supported (${builtins.toJSON finalAttrs.passthru.supportedRedistSystems})";
+              assertion = isSupportedRedistSystem;
+            }
+          ]
+          ++ platformAssertions;
+      };
+
+      meta = meta // {
+        longDescription = meta.longDescription or "" + ''
+          By downloading and using this package you accept the terms and conditions of the associated license(s).
+        '';
+        sourceProvenance = meta.sourceProvenance or [ sourceTypes.binaryNativeCode ];
+        platforms = finalAttrs.passthru.supportedNixSystems;
+        broken = _mkMetaBroken finalAttrs;
+        badPlatforms = _mkMetaBadPlatforms finalAttrs;
+        downloadPage =
+          meta.downloadPage
+            or "https://developer.download.nvidia.com/compute/${finalAttrs.passthru.redistName}/redist/${finalAttrs.pname}";
+        # NOTE:
+        #   Every redistributable should set its own license; since that's a lot of manual work, we default to
+        #   nvidiaCudaRedist if the redistributable is from the CUDA redistributables and nvidiaCuda otherwise.
+        #   Despite calling them "redistributable" and the download URL containing "redist", a number of these
+        #   packages are not licensed such that redistribution is allowed.
+        license =
+          if meta ? license then
+            lib.toList meta.license
+          else if finalAttrs.passthru.redistName == "cuda" then
+            [ licenses.nvidiaCudaRedist ]
+          else
+            [ licenses.nvidiaCuda ];
+        teams = meta.teams or [ ] ++ [ teams.cuda ];
```

Comment:
Make sure teams aren't specified in the redist packages with teams.cuda so we don't end up with double entries for the team.